### PR TITLE
manifest: update nrf_modem build system files

### PR DESCRIPTION
**Pull request by lemrey**
Bring in build system updates.

### CODE REVIEW

File: west.yml [hunk 1]
```diff
@@ -141,7 +141,7 @@ manifest:
     - name: nrfxlib
       repo-path: sdk-nrfxlib
       path: nrfxlib
-      revision: 15307eb68e1d465550c80f8dd81e0b12ae44a78c
+      revision: pull/1089/HEAD
```
Comment (rlubos) [inline west.yml:144]: You need to use lowercase `head`

Comment (lemrey) [reply west.yml]: Thank you

**Comment by lemrey** [comment]
I do not think this change is visible to the user so I removed the `changelog-entry-required` label.

**Review by eivindj-nordic — APPROVED** [review]

File: west.yml [hunk 2]
```diff
@@ -141,7 +141,7 @@ manifest:
     - name: nrfxlib
       repo-path: sdk-nrfxlib
       path: nrfxlib
-      revision: f8ceac162de19cbcee9652cb428b217b1cf85ab3
+      revision: pull/1089/head
```
Comment (rlubos) [inline west.yml:144]: `6db09621d01fd246d21010adc6ac8bdcb2a0e16a`

Comment (lemrey) [reply west.yml]: Thank you

**Review by MirkoCovizzi — APPROVED** [review]

**Pull request merged**